pull_request_template.md

```diff
@@ -1,33 +1,60 @@
+FIX #xxxx (*link existing issues this PR will resolve*)
 
-Fixes # (issue)
+**BEFORE SUBMITTING, PLEASE READ THE CHECKLIST BELOW AND FILL IN THE DESCRIPTION ABOVE**
 
-## Type of change
+---
 
-Please delete options that are not relevant.
+<details>
+<!-- inside this <details> section, markdown rendering does not work, so we use raw html here. -->
+<summary><b> PR Checklist (Click to Expand) </b></summary>
 
-- [ ] Bug fix (non-breaking change which fixes an issue)
-- [ ] New feature (non-breaking change which adds functionality)
-- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-- [ ] This change requires a documentation update
+<br>
 
-# How Has This Been Tested?
+<p>Thank you for your contribution to gigax! Before submitting the pull request, please ensure the PR meets the following criteria. This helps gigax maintain the code quality and improve the efficiency of the review process.</p>
 
-Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
+<h3>PR Title and Classification</h3>
+<p>Only specific types of PRs will be reviewed. The PR title is prefixed appropriately to indicate the type of change. Please use one of the following:</p>
+<ul>
+    <li><code>[Bugfix]</code> for bug fixes.</li>
+    <li><code>[CI/Build]</code> for build or continuous integration improvements.</li>
+    <li><code>[Doc]</code> for documentation fixes and improvements.</li>
+    <li><code>[Model]</code> for adding a new model or improving an existing model. Model name should appear in the title.</li>
+    <li><code>[Core]</code> for changes in the core gigax logic</li>
+    <li><code>[Misc]</code> for PRs that do not fit the above categories. Please use this sparingly.</li>
+</ul>
+<p><strong>Note:</strong> If the PR spans more than one category, please include all relevant prefixes.</p>
 
-- [ ] Ran unit tests on local environment
-- [ ] Deployed local version of our app, and tested the feature in-game
+<h3>Code Quality</h3>
 
+<p>The PR need to meet the following code quality standards:</p>
 
-# Checklist:
+<ul>
+    <li>We adhere to <a href="https://google.github.io/styleguide/pyguide.html">Google Python style guide</a> and <a href="https://google.github.io/styleguide/cppguide.html">Google C++ style guide</a>.</li>
+    <li>The code need to be well-documented to ensure future contributors can easily understand the code.</li>
+    <li>Include sufficient tests to ensure the project to stay correct and robust. This includes both unit tests and integration tests.</li>
+    <li>Please add documentation to <code>docs/source/</code> if the PR modifies the user-facing behaviors of gigax. It helps gigax user understand and utilize the new features or changes.</li>
```

Comment:
No `docs/source` yet, because our docs are on the docusaurus frontend; but it would be great to add the docs directly to the repo here! This would mean: API docs generated automatically from GitHub code, whenever we push to this repo :) 
This is what vllm are doing with read-the-docs; and Outlines with [mkdocs (material theme)](https://squidfunk.github.io/mkdocs-material/). 
What do you think?
```suggestion
```

Reply:
Ok, great idea, how we can organize ourselves to achieve this.

Reply:
I'll do it whenever I have the time :) 
[GIG-185 : Replace docusaurus with Mkdocs page](https://linear.app/gigax/issue/GIG-185/replace-docusaurus-with-mkdocs-page)